src/display/api.js

```diff
@@ -994,16 +1082,224 @@ var PDFPageProxy = (function PDFPageProxyClosure() {
 })();
 
 /**
+ * PDF.js web worker abstraction, it controls instantiation of PDF documents and
+ * WorkerTransport for them.  If creation of a web worker is not possible,
+ * a "fake" worker will be used instead.
+ * @class
+ */
+var PDFWorker = (function PDFWorkerClosure() {
+  var fakeWorkerReady = null;
+  var nextFakeWorkerId = 0;
+
+  // Loads worker code into main thread and setups up the port we will be using
+  // in MessageHandler instances.
+  function setupFakeWorkerGlobal() {
```

Comment:
Alot of this is coming from old code, but some of the new parts could be flattened. I don't think there's a reason to create a new promise. Something like https://pastebin.mozilla.org/8852994, in subsequent pull we could make this even better by getting rid of the capability promise.